### PR TITLE
Updated varaibles cavh

### PR DESCRIPTION
**Pull request by jfallmann**
More fixes and changes from @cavelandiah

### CODE REVIEW

File: scripts/MIRfix.py [hunk 1]
```diff
@@ -257,7 +259,7 @@ def flip(filename,filen,outdir,mappingfile,matfile,listofnew,listofnewloop,listo
                                 log.debug(["mat desc is here",matdesc])
                                 matseq=str(mat.seq).strip()
                                 matseq=matseq.replace('T','U')#replace T by U because the RNAfold produces the sequences as U
-                                spos=str(precseq).index(matseq)
+                                spos=str(precseq).find(matseq)
```
Comment (jfallmann) [inline scripts/MIRfix.py:262]: index should be safer here

File: scripts/MIRfix.py [hunk 2]
```diff
@@ -276,7 +278,7 @@ def flip(filename,filen,outdir,mappingfile,matfile,listofnew,listofnewloop,listo
                 elif xcut<ycut and xcut<=10:#=> 5p and no need to cut, already <=10
                     precseq=precseq
 
-                spos=str(precseq).index(matseq)#spos after cut
+                spos=str(precseq).find(matseq)#spos after cut
```
Comment (jfallmann) [inline scripts/MIRfix.py:281]: here as well

File: scripts/MIRfix.py [hunk 3]
```diff
@@ -636,8 +638,8 @@ def readfold(listnewold,filename,oldlstlstr,oldlstlstl,spos,epos,newspos,newepos
                         newparts=parts
                     for i in range(0,len(partslist)):
                         if i==0 and partslist[0]!=-1:#if the partslist starting from position zero, i.e. the first part starts at position zero
-                            loopstart=hairpin.rindex('(',0,int(partslist[0])+1)+1
-                            loopend=hairpin.index(')',0,int(partslist[0])+1)-1
+                            loopstart=hairpin.rfind('(',0,int(partslist[0])+1)+1
```
Comment (jfallmann) [inline scripts/MIRfix.py:641]: and here

File: scripts/MIRfix.py [hunk 4]
```diff
@@ -636,8 +638,8 @@ def readfold(listnewold,filename,oldlstlstr,oldlstlstl,spos,epos,newspos,newepos
...
+                            loopend=hairpin.find(')',0,int(partslist[0])+1)-1
```
Comment (jfallmann) [inline scripts/MIRfix.py:642]: and here

File: scripts/MIRfix.py [hunk 5]
```diff
@@ -725,8 +727,8 @@ def readfold(listnewold,filename,oldlstlstr,oldlstlstl,spos,epos,newspos,newepos
 
 
                         if i==len(partslist)-1 and partslist[i]!=-1:
-                            loopstart=hairpin.rindex('(',int(partslist[i])+1)+1
-                            loopend=hairpin.index(')',int(partslist[i])+1)-1
+                            loopstart=hairpin.rfind('(',int(partslist[i])+1)+1
```
Comment (jfallmann) [inline scripts/MIRfix.py:730]: and here

File: scripts/MIRfix.py [hunk 6]
```diff
@@ -725,8 +727,8 @@ def readfold(listnewold,filename,oldlstlstr,oldlstlstl,spos,epos,newspos,newepos
...
+                            loopend=hairpin.find(')',int(partslist[i])+1)-1
```
Comment (jfallmann) [inline scripts/MIRfix.py:731]: and here

File: scripts/MIRfix.py [hunk 7]
```diff
@@ -812,8 +814,8 @@ def readfold(listnewold,filename,oldlstlstr,oldlstlstl,spos,epos,newspos,newepos
                                 currcounts=int(hairpin[int(partslist[i]):].count("("))+int(hairpin[int(partslist[i]):].count(")"))
 
                         if i!=0 and i!=len(partslist)-1 and i%2==0:
-                            loopstart=hairpin.rindex('(',int(partslist[i-1]),int(partslist[i])+1)+1
-                            loopend=hairpin.index(')',int(partslist[i-1]),int(partslist[i])+1)-1
+                            loopstart=hairpin.rfind('(',int(partslist[i-1]),int(partslist[i])+1)+1
```
Comment (jfallmann) [inline scripts/MIRfix.py:817]: and here

File: scripts/MIRfix.py [hunk 8]
```diff
@@ -812,8 +814,8 @@ def readfold(listnewold,filename,oldlstlstr,oldlstlstl,spos,epos,newspos,newepos
...
+                            loopend=hairpin.find(')',int(partslist[i-1]),int(partslist[i])+1)-1
```
Comment (jfallmann) [inline scripts/MIRfix.py:818]: and here

**Review by cavelandiah — APPROVED** [review]

File: scripts/MIRfix.py [hunk 9]
```diff
@@ -1057,8 +1059,8 @@ def readfold(listnewold,filename,oldlstlstr,oldlstlstl,spos,epos,newspos,newepos
                         finalnewcomp=finalcomp
 
                 if parts==1:
-                    loopstart=hairpin.rindex('(')+1
-                    loopend=hairpin.index(')')-1
+                    loopstart=hairpin.rfind('(')+1
```
Comment (jfallmann) [inline scripts/MIRfix.py:1062]: if no structure is found we also want an error

Comment (jfallmann) [reply scripts/MIRfix.py:1062]: if no structure is found we also want an error

File: scripts/MIRfix.py [hunk 10]
```diff
@@ -2676,13 +2683,13 @@ def readfoldpredict(foldfile,predictedspos,predictedepos):#,predictedspos,predic
         for line in SeqIO.parse(fofi,"fasta"):
             string=str(line.seq)
             if ((")") or ("(") or (".")) in str(line.seq):#to get where the folding starts, ex1: ACGTtgatagt..((..))(score) ex2: acgtatgat(((.)))(socre)
-                rindex=string.index(")")
-                lindex=string.index("(")
-                pindex=string.index(".")
+                rindex=string.find(")")
```
Comment (jfallmann) [inline scripts/MIRfix.py:2686]: if no structure is found we also want an error

File: scripts/MIRfix.py [hunk 11]
```diff
@@ -3188,29 +3195,29 @@ def correct(corid,flanking,countcorrected,countcorrectedTonew,listofnew,listofne
                 correctmir=str(listoldstatus[corrind+3])
                 correctmirstar=str(listoldstatus[corrind+4])
                 orien=str(listoldstatus[corrind+5])
-                coor1=int(longseq.index(correctmir))
-                coor2=int(longseq.index(correctmirstar))
+                coor1=int(longseq.find(correctmir))
```
Comment (jfallmann) [inline scripts/MIRfix.py:3198]: here index should be fine as we have to find the mir, if its missing we want an error

File: scripts/MIRfix.py [hunk 12]
```diff
@@ -3188,29 +3195,29 @@ def correct(corid,flanking,countcorrected,countcorrectedTonew,listofnew,listofne
...
+                coor2=int(longseq.find(correctmirstar))
 
                 if coor2<coor1:
                     tempcor=correctmir[:]
                     correctmir=correctmirstar[:]
                     correctmirstar=tempcor[:]
 
-                startmatlong=int(longseq.index(correctmir))#position of mir in the long seq
+                startmatlong=int(longseq.find(correctmir))#position of mir in the long seq
```
Comment (jfallmann) [inline scripts/MIRfix.py:3206]: same here

File: scripts/MIRfix.py [hunk 13]
```diff
@@ -3624,13 +3631,13 @@ def sublist(filename):
                     for record in SeqIO.parse(mtf, 'fasta'):
                         curmatseq=str(record.seq)
                         if firstmat.strip() in record.description:
-                            startmat=int(mat2seq.index(curmatseq))
+                            startmat=int(mat2seq.find(curmatseq))
```
Comment (jfallmann) [inline scripts/MIRfix.py:3634]: if not found we want an error

**Review by jfallmann — COMMENTED** [review]
@cavelandiah  We need to fix some issues here, please see the comments

**Pull request merged**